### PR TITLE
Update mdstat parser to remove asserts

### PR DESCRIPTION
**Pull request by ryan-blakley**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:
Went through and replaced all asserts with if statements that raise parse exceptions if not met. This required several logic updates due to how the asserts were used. Updated the doc strings of each function so they were uniform. Updated the test to check for parse exceptions instead of assertion errors, also added a couple additional test.

### CODE REVIEW

File: insights/parsers/mdstat.py [hunk 1]
```diff
@@ -154,107 +157,118 @@ def parse_content(self, content):
...
 
     for token in tokens:
-        subtokens = re.split(r"\[|]", token)
-        assert len(subtokens) > 1
-        comp_name = subtokens[0]
-        assert comp_name
+        # Token should be sda1[0] or sda1[0](S) for
+        # example, and subtokens should be
+        # ['sda1', '0'] or ['sda1', '0', '(S)'].
+        subtokens = re.split(r"[\[\]]", token)
```
Comment (bfahr) [inline insights/parsers/mdstat.py:251]: Is there a neg/pos test that covers subtokens, I wasn't able to find it.

**Review by bfahr — CHANGES_REQUESTED** [review]

File: insights/parsers/mdstat.py [hunk 2]
```diff
@@ -1,106 +1,104 @@
...
+              129596288 blocks [2/2] [UU]
+
+        md3 : active raid5 sdl1[9] sdk1[8] sdj1[7] sdi1[6] sdh1[5] sdg1[4] sdf1[3] sde1[2] sdd1[1] sdc1[0]
+              1318680576 blocks level 5, 1024k chunk, algorithm 2 [10/10] [UUUUUUUUUU]
+
+    The data contained in ``mdstat`` is represented with three top level members -
+    ``personalities``, ``components`` and ``mds``.
+
+    Attributes:
+        personalities:
```
[the remark this answers is not in the document]
Comment (ryan-blakley) [reply insights/parsers/mdstat.py]: Whoops I went through and fixed this.

File: insights/parsers/mdstat.py [hunk 3]
```diff
@@ -1,106 +1,104 @@
...
+
+             - ``name``: Name of the MD device
+             - ``active``: Whether the MD device is active
+             - ``raid``: The RAID type string
+             - ``devices``: a list of the devices in this
+             - ``blocks``, ``level``, ``chunk`` and ``algorithm`` - the same
+               information given above per component device (if found)
+
+    Examples:
+        >>> mdstat = shared[Mdstat]
```
[the remark this answers is not in the document]
Comment (ryan-blakley) [reply insights/parsers/mdstat.py]: I updated it and changed it to using type.

File: insights/parsers/mdstat.py [hunk 4]
```diff
@@ -154,107 +156,118 @@ def parse_content(self, content):
 
 
 def parse_personalities(personalities_line):
-    """Parse the "personalities" line of ``/proc/mdstat``.
+    """
+    Parse the "personalities" line of ``/proc/mdstat``.
 
-    Lines are expected to be like:
+    Sample of personalities_line:
```
[the remark this answers is not in the document]
Comment (ryan-blakley) [reply insights/parsers/mdstat.py]: I went through and fixed this on each sample.

File: insights/parsers/mdstat.py (hunk 1 above)
Comment (ryan-blakley) [reply insights/parsers/mdstat.py:249]: It's in the test_parse_array_start function in the MD_FAIL test from what I can tell, let me know if I interrupted that test incorrectly.

**Pull request merged**